File: app/src/main/java/com/wordpress/historiaspassageiras/shoppinglist/Item.java
package com.wordpress.historiaspassageiras.shoppinglist;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.Gson;

public class Item implements Parcelable {
    public static final Parcelable.Creator<Item> CREATOR
            = new Parcelable.Creator<Item>() {
        public Item createFromParcel(Parcel in) {
            return new Item(in);
        }

        public Item[] newArray(int size) {
            return new Item[size];
        }
    };
    private String item;
    private Boolean done;

    public Item(String i, Boolean d) {
        item = i;
        done = d;
    }

    private Item(Parcel in) {
        item = in.readString();
        done = in.readInt() == 1;
    }

    static public Item create(String seralizedData) {
        Gson gson = new Gson();
        return gson.fromJson(seralizedData, Item.class);
    }

    void SetItem(String i) {
        item = i;
    }

    String GetItem() {
        return item;
    }

    void Done() {
        done = true;
    }

    void Undone() {
        done = false;
    }

    Boolean IsDone() {
        return done;
    }

    public int describeContents() {
        return 0;
    }

    public void writeToParcel(Parcel out, int flags) {
        out.writeString(item);
        out.writeInt(done ? 0 : 1);
    }

    public String serialize() {
        Gson gson = new Gson();
        return gson.toJson(this);
    }
}

File: app/src/main/java/com/wordpress/historiaspassageiras/shoppinglist/Item.kt
package com.wordpress.historiaspassageiras.shoppinglist

import android.os.Parcel
import android.os.Parcelable

import com.google.gson.Gson

class Item : Parcelable {
    private var item: String = ""
    private var itemDone: Boolean = false

    constructor(i: String, d: Boolean) {
        item = i
        itemDone = d
    }

    private constructor(`in`: Parcel) {
        item = `in`.readString()
        itemDone = `in`.readInt() == 1
    }

    internal fun getItem() = item

    internal fun done() {
        itemDone = true
    }

    internal fun undone() {
        itemDone = false
    }

    internal fun isDone() = itemDone

    override fun describeContents() = 0

    override fun writeToParcel(out: Parcel, flags: Int) {
        out.writeString(item)
        out.writeInt(if (itemDone) 0 else 1)
    }

    fun serialize(): String = Gson().toJson(this)

    companion object CREATOR : Parcelable.Creator<Item> {
        override fun createFromParcel(parcel: Parcel): Item {
            return Item(parcel)
        }

        override fun newArray(size: Int): Array<Item?> {
            return arrayOfNulls(size)
        }

        fun create(serializedData: String): Item {
            return  Gson().fromJson(serializedData, Item::class.java)
        }
    }
}